packages/iov-ethereum/src/ethereumcodec.ts

```diff
@@ -26,10 +34,76 @@ export function toChecksumAddress(address: string): Address {
 
 export const ethereumCodec: TxCodec = {
   bytesToSign: (unsigned: UnsignedTransaction, nonce: Nonce): SigningJob => {
-    throw new Error(`Not implemented utx: ${unsigned}, nonce: ${nonce}`);
+    return {
+      bytes: Serialization.serializeTransaction(unsigned, nonce) as SignableBytes,
+      prehashType: PrehashType.Keccak256,
+    };
   },
   bytesToPost: (signed: SignedTransaction): PostableBytes => {
-    throw new Error(`Not implemented tx: ${signed}`);
+    switch (signed.transaction.kind) {
+      case TransactionKind.Send:
+        let gasPriceHex = "0x";
+        let gasLimitHex = "0x";
+        let dataHex = "0x";
+        let nonceHex = "0x";
+
+        const valueHex = encodeQuantityString(
+          Serialization.amountFromComponents(
+            signed.transaction.amount.whole,
+            signed.transaction.amount.fractional,
+          ),
+        );
+        if (signed.primarySignature.nonce.toNumber() > 0) {
+          nonceHex = encodeQuantity(signed.primarySignature.nonce.toNumber());
+        }
+        if (signed.transaction.gasPrice) {
+          gasPriceHex = encodeQuantityString(
+            Serialization.amountFromComponents(
+              signed.transaction.gasPrice.whole,
+              signed.transaction.gasPrice.fractional,
+            ),
+          );
+        }
+        if (signed.transaction.gasLimit) {
+          gasLimitHex = encodeQuantityString(
+            Serialization.amountFromComponents(
+              signed.transaction.gasLimit.whole,
+              signed.transaction.gasLimit.fractional,
+            ),
+          );
+        }
+        if (signed.transaction.memo) {
+          dataHex += Encoding.toHex(Encoding.toUtf8(signed.transaction.memo));
+        }
+        if (!isValidAddress(signed.transaction.recipient)) {
+          throw new Error("Invalid recipient address");
+        }
+        const sig = signed.primarySignature.signature;
+        const r = sig.slice(0, 32);
+        const s = sig.slice(32, 64);
+        let v = Number(sig.slice(-1)) + 27;
+        const chainId = Number(signed.transaction.chainId);
+        if (chainId > 0) {
+          v += chainId * 2 + 8;
+        }
+        const chainIdHex = encodeQuantity(v);
+        const postableTx = new Uint8Array(
+          toRlp([
+            Buffer.from(fromHex(hexPadToEven(nonceHex))),
+            Buffer.from(fromHex(hexPadToEven(gasPriceHex))),
+            Buffer.from(fromHex(hexPadToEven(gasLimitHex))),
+            Buffer.from(fromHex(hexPadToEven(signed.transaction.recipient))),
+            Buffer.from(fromHex(hexPadToEven(valueHex))),
+            Buffer.from(fromHex(hexPadToEven(dataHex))),
+            Buffer.from(fromHex(hexPadToEven(chainIdHex))),
+            Buffer.from(fromHex(trimLeadingZero(toHex(r)))),
+            Buffer.from(fromHex(trimLeadingZero(toHex(s)))),
```

Comment:
`r` and `s` from ExtendedSecp256k1Signature have no leading 0s. So this becomes just

```
Buffer.from(r),
Buffer.from(s),
```

and `trimLeadingZero` can be removed